### PR TITLE
Add XML command "GetLifeSpan"

### PR DESCRIPTION
**Pull request by flubshi**
Adds GetLifeSpan command.

Compared to the JSON commands, this allows to request only one component_type per request.

JSON life span types always start with a lower letter, while XML compnent type always starts capitalized. I am currently not really happy with my approach of converting these types.

Runtime tested:

image

## Summary by CodeRabbit

- **New Features**
  - Introduced a new command, `GetLifeSpan`, for handling life span-related XML messages.
  - Added a new lifespan component, `DUST_CASE_HEAP`, to the existing `LifeSpan` enum.

- **Bug Fixes**
  - Enhanced error handling for the `GetLifeSpan` command with improved response validation.

- **Tests**
  - Added unit tests for the `GetLifeSpan` command to ensure proper functionality and error handling.

- **Refactor**
  - Updated the `has_sub_element` method to be an instance-level property in the `XmlCommand` class.

### CODE REVIEW

File: deebot_client/commands/xml/life_span.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+class GetLifeSpan(XmlCommandWithMessageHandling):
+    """GetLifeSpan command."""
+
+    name = "GetLifeSpan"
+
+    def __init__(self, life_span: LifeSpan) -> None:
+        # type for JSON commands starts with small letter, while XML types start with upper letter. Workaround:
+        xml_type = life_span.value[0].upper() + life_span.value[1:]
+        super().__init__({"type": xml_type})
```
[the remark this answers is not in the document]
Comment (flubshi) [reply deebot_client/commands/xml/life_span.py:26]:
```
>>> "thisDoesNotWork".capitalize()
'Thisdoesnotwork' <> 'ThisDoesNotWork'
```

File: deebot_client/commands/xml/life_span.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                and "total" in xml.attrib):
+
+            component_type = xml.attrib.get("type")
+            # type for JSON commands starts with small letter, while XML types start with upper letter. Workaround:
+            xml_type = component_type[0].lower() + component_type[1:]
+            left = int(xml.attrib.get("left"))
+            total = int(xml.attrib.get("total"))
+
+            percent = round((left / total) * 100, 2)
+            event_bus.notify(LifeSpanEvent(LifeSpan(xml_type), percent, left))
```
[the remark this answers is not in the document]
Comment (flubshi) [reply deebot_client/commands/xml/life_span.py:46]: I prefer the original code, because it is better human readable

**Review by edenhaus — CHANGES_REQUESTED** [review]
Can you please fix the CI

**Comment by flubshi** [comment]
pre-commit.ci autofix

**Comment by flubshi** [comment]
> Can you please fix the CI

@edenhaus I fixed everything except for one mypy issue. Could you please help here? Feel free to change directly

**Comment by flubshi** [comment]
@edenhaus CI is happy now. Only failing code quality job seems unrelated, because dev is also affected. What are the next steps to merge?

**Comment by edenhaus** [comment]
Will review it in the next days :)

**Review by edenhaus — APPROVED** [review]
Thanks @flubshi :+1:

**Pull request merged**